langchain/llms/databricks_endpoint.py

```diff
@@ -0,0 +1,302 @@
+    """
+
+    transform_input_fn: Optional[Callable] = None
+    """A function that transforms ``(prompt, stop)`` into a JSON-compatible request
+    that the endpoint accepts.
+    For example, you can insert additional parameters like temperature.
+    """
+
+    transform_output_fn: Optional[Callable[..., str]] = None
+    """A function that transforms the output from the endpoint to the generated text.
+    """
+
+    _client: DatabricksClientBase = PrivateAttr()
+
+    class Config:
+        extra = Extra.forbid
+        underscore_attrs_are_private = True
+
+    @validator("cluster_id", always=True)
+    def set_cluster_id(cls, v: Any, values: Dict[str, Any]) -> Optional[str]:
+        if values["endpoint_name"]:
+            assert v is None, "Cannot provide both endpoint_name and cluster_id."
+        elif not v:
+            try:
+                v = get_repl_context().clusterId
+            except Exception:
+                pass
+            assert v, (
+                "Neither endpoint_name nor cluster_id was set. "
+                "And the cluster_id cannot be automatically determined."
+            )
+        return v
+
+    @validator("cluster_driver_port", always=True)
+    def set_cluster_driver_port(cls, v: Any, values: Dict[str, Any]) -> Optional[str]:
+        if values["endpoint_name"]:
+            assert v is None, "Cannot set both endpoint_name and cluster_driver_port."
+        else:
+            assert (
+                v is not None
+            ), "Must set cluster_driver_port to connect to a cluster driver."
+            assert int(v) > 0, f"Invalid cluster_driver_port: {v}"
+        return v
+
+    def __init__(self, **data: Any) -> None:
+        super().__init__(**data)
+        if self.endpoint_name:
+            self._client = DatabricksServingEndpointClient(
+                host=self.host,
+                api_token=self.api_token,
+                endpoint_name=self.endpoint_name,
+            )
+        elif self.cluster_id and self.cluster_driver_port:
+            self._client = DatabricksClusterDriverProxyClient(
+                host=self.host,
+                api_token=self.api_token,
+                cluster_id=self.cluster_id,
+                cluster_driver_port=self.cluster_driver_port,
+            )
+        else:
+            raise ValueError(
+                "Must specify either endpoint_name or cluster_id/driver_port."
+            )
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "databricks_endpoint"
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+    ) -> str:
+        """Queries the LLM endpoint with the given prompt and stop sequence."""
+
+        # TODO: support callbacks
+
+        request = {"prompt": prompt, "stop": stop}
+        if self.transform_input_fn:
```

Comment:
will transform_input_fn usually just be adding kwargs or actually changing existing keys? if it's the former thoughts on replacing `transform_input_fn` with `model_kwargs: dict = Field(default_factory=dict)` and here doing
```python
request = {"prompt": prompt, "stop": stop, **self.model_kwargs}
```

Reply:
or could have both

Reply:
It could change existing keys, depending how the model was served. So let's have both :)